arbos/internal_tx.go

```diff
@@ -61,19 +61,23 @@ func ApplyInternalTxUpdate(tx *types.ArbitrumInternalTx, state *arbosState.Arbos
 			// (incorrectly) use the L2 block number instead
 			timePassed = util.SafeMapGet[uint64](inputs, "l2BlockNumber")
 		}
+		if state.ArbOSVersion() < 8 {
+			// in old versions we incorrectly used an L1 block number one too high
+			l1BlockNumber++
+		}
 
 		nextL1BlockNumber, err := state.Blockhashes().NextBlockNumber()
 		state.Restrict(err)
 
 		l2BaseFee, err := state.L2PricingState().BaseFeeWei()
 		state.Restrict(err)
 
-		if l1BlockNumber >= nextL1BlockNumber {
+		if l1BlockNumber > nextL1BlockNumber {
```

Comment:
We should arguably rename `nextL1BlockNumber` (and the corresponding things in the blockhash code) to `currentL1BlockNumber`.